public/js/test/mochitest/browser_dbg-editor-select.js

```diff
@@ -0,0 +1,68 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+// Tests that the editor highlights the correct location when the
+// debugger pauses
+
+const TAB_URL = EXAMPLE_URL + "doc-scripts.html";
+
+function assertPausedLocation(dbg, source, line) {
+  const { selectors: { getSelectedSource, getPause }, getState } = dbg;
+
+  is(getSelectedSource(getState()).get("url"), source.url);
+  const location = getPause(getState()).getIn(["frame", "location"]);
+  is(location.get("sourceId"), source.id);
+  is(location.get("line"), line);
+}
+
+add_task(function* () {
+  const dbg = yield initDebugger(
+    TAB_URL,
+    "simple1.js", "simple2.js", "long.js"
+  );
+  const { selectors: { getSelectedSource }, getState } = dbg;
+  const simple1 = findSource(dbg, "simple1.js");
+  const simple2 = findSource(dbg, "simple2.js");
+
+  // Set the initial breakpoint.
+  yield addBreakpoint(dbg, simple1.id, 4);
+  ok(!getSelectedSource(getState()), "No selected source");
+
+  // Call the function that we set a breakpoint in.
+  ContentTask.spawn(gBrowser.selectedBrowser, null, function* () {
+    content.wrappedJSObject.main();
+  });
+
+  // Make sure that it pauses in the expected source.
+  yield waitForPaused(dbg);
+  is(getSelectedSource(getState()).get("url"), simple1.url);
+
+  // Step through to another file and make sure it's paused in the
+  // right place.
+  yield stepIn(dbg);
+  assertPausedLocation(dbg, simple2, 2);
+
+  // Step back out to the initial file.
+  yield stepOut(dbg);
+  yield stepOut(dbg);
+  assertPausedLocation(dbg, simple1, 5);
```

Comment:
Maybe, but I'd like to land 5-10 mochitests now and we can flesh them out/add more as we get a good feeling for it. I'd like to do it breadth-first, because writing several different kinds of mochitests will give us a feeling for what should be actually shared across them, and how we should structure them.